### ISSUE TITLE
AddrAdd, RouteAdd, and potentially others fail when IPNet Mask is derived from ParseIP

### ISSUE THREAD

**Issue by matt-johnston-ecommistry** [vishvananda/netlink#576]
If the Mask passed into an Address or Route are in 16-byte format, for example if they are generated with net.IPMask(net.ParseIP("255.255.255.0")), then unintended behavior occurs.
Running String() on the generated IPNet still shows the correct result - for example:
	addr = &netlink.Addr{
			IPNet: &net.IPNet{
				IP:   net.ParseIP("172.31.0.1"),
				Mask: net.IPMask(net.ParseIP("255.255.255.0")),
			},
		}

This will throw an index out of range panic, on the Broadcast address generation. If Broadcast address is manually specified, then this will create the interface successfully, but as 172.31.0.1/0 instead of 172.31.0.1/24.

For RouteAdd, this will fail with "invalid argument".

I believe other calls are similarly affected, but was not able to test them all.

In either case, running String() on the IPNet shows the correct /24 suffix, as Go's code seems to transparently handle either 4-byte or 16-byte (v4-in-v6) format.

The fix from the user end is to do Mask: net.IPMask(net.ParseIP("255.255.255.0").To4())
Or alternatively use the actual net.IPv4Mask function, however this is not always easy if you need to pass in a string.

That said, it would be ideal if the netlink library here could identify when the IPNet it's passed is the wrong length and provide a sand behavior - either a len check based on the requested family throwing a descriptive error, or just transparently calling To4(). Either behavior would be better than creating invalid interface IPs.

This has been reproduced on Go 1.15, compiling for Linux, however the code appears to not handle this in any environment.

**Comment by ymmt2005** [comment]
I think the right way to create a mask is to use either [`net.CIDRMask`](https://pkg.go.dev/net#CIDRMask) or [`net.IPv4Mask`](https://pkg.go.dev/net#IPv4Mask).

**Comment by matt-johnston-ecommistry** [comment]
Yeah, I agree - the documentation however isn't 100% clear on that (which I'm well aware is more of an upstream Go issue).

My concern here is that this library shouldn't panic, nor create malformed Address entries or Routes on something which is a very easy validation (len(mask) == 4 for example).
